tests/utils.ts

```diff
@@ -25,6 +25,15 @@ export function outputForInput(
   const dir = dirname(input);
   let stem = basename(input, extname(input));
 
+  // TODO: there's a bug where output-ext keys from a custom format are 
```

Comment:
Hey @dragonstyle can you think of a clean way to fix this? This hack looks super ugly, but a real fix seems to need access to metadata in a way that would complicate things quite a bit.

Reply:
Just chiming in because I tweaked this a bit when adding typst support. `output-ext` key is recognized when passed as metadata in the yaml of the smoke test .qmd file. 
like doing  
````
format: 
  confluence-publish:
    output-ext: xml
```` 

My understanding at the time is that extension is not derived from `output-ext` value from a format, but computed from `baseFormat`. (which would be publish here). I don't think internal `output-ext` value from other non-custom format is taken into account either. 

So I think your workaround around is a good one. Otherwise, it would be like other format by adding a `if (formatDesc.baseFormat === "publish") {` handler. Not that clear. 

I may be missing something. Personally I am even fine with explicitly adding this to our test .qmd file 😅 
````yaml
format: 
  confluence-publish:
    output-ext: xml
```` 

Anyhow, I may be missing something though... 🤷‍♂️

Reply:
> Just chiming in because I tweaked this a bit when adding typst support. `output-ext` key is recognized when passed as metadata in the yaml of the smoke test .qmd file. like doing
> 
> ```
> format: 
>   confluence-publish:
>     output-ext: xml
> ```

Hm. That might be the case, but we don't actually have any tests that use this syntax. If you search for `output-ext` in the repository, the only .qmd files that match are:

- tests/docs/smoke-all/2023/09/27/7041.qmd
- tests/docs/smoke-all/typst/typst-citeproc.qmd
- tests/docs/smoke-all/typst/typst-no-citeproc.qmd

All of these documents use `output-ext: typ` in the top-level of the metadata, and not in the test spec (and it changes the behavior of the rendering, not just the extension, like we discussed before)

Reply:
I'm just going to merge this hack as is, and we can resolve the situation later.